src/sentry/api/endpoints/data_export.py

```diff
@@ -0,0 +1,46 @@
+from __future__ import absolute_import
+
+from sentry import features
+from sentry.api.bases.incident import IncidentPermission
+from sentry.api.base import Endpoint
+from django.utils import timezone
+
+
+from sentry.tasks.data_export import create_record
+
+from sentry.models import ExportedData, Organization
+
+
+class DataExportEndpoint(Endpoint):
+    permission_classes = (IncidentPermission,)
+
+    def get(self, *args, **kwargs):
+        """
+        Retrieve information about the temporary file record.
+        Used to populate page emailed to the user.
+        """
+        organization = Organization.objects.get(slug=kwargs["organization_slug"])
+        if not features.has("organizations:data-export", organization):
+            return self.respond(status=404)
+        try:
+            export_record = ExportedData.objects.get(data_id=kwargs["data_id"])
+            if export_record.finished_at is None:
+                return self.respond({"status": "EARLY"})
+            elif export_record.expired_at < timezone.now():
+                return self.respond({"status": "EXPIRED"})
+            else:
+                return self.respond(export_record.get_storage_info())
+        except ExportedData.DoesNotExist:
+            return self.respond({"status": "INVALID"})
+
+    def post(self, request, *args, **kwargs):
```

Comment:
We typically put `post` endpoints in a separate endpoint to `get/put/delete`. This is because your url for this endpoint involves a `data_id`, but you have no id when actually creating the row initially. Usually we'll put this in an endpoint like `DataExportListEndpoint`